### ISSUE TITLE
Issue Running miRge_build on Singularity Image

### ISSUE THREAD

**Issue by TomokiMotegi** [mhalushka/miRge3_build#7]
Hello,

I’ve been working on creating an original miRNA library for dogs. I successfully generated the necessary FASTA files, but I encountered an issue when attempting to install miRge_build locally. To work around this, I tried running miRge_build using the Singularity image available from BioContainer:
https://biocontainers.pro/tools/mirge-build
However, I encountered the following error during the process:
```
bowtie version: 1.2.3

Library indexing in progress...

Traceback (most recent call last):
  File "/usr/local/bin/miRge-build", line 10, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.7/site-packages/mirge_build/__main__.py", line 51, in main
    bld_bowtie_index(args, str(index_path), str(bwtCommand2))
  File "/usr/local/lib/python3.7/site-packages/mirge_build/blibs/miRge_bowtie_build.py", line 40, in bld_bowtie_index
    newSeq = miRNA_miRge_format(hpin_mir_name[name], sequence)
  File "/usr/local/lib/python3.7/site-packages/mirge_build/blibs/miRge_bowtie_build.py", line 12, in miRNA_miRge_format
    start, end = re.search(mir, hpin).span()
AttributeError: 'NoneType' object has no attribute 'span'
```
It seems the current Singularity image uses Python 3.7.8, whereas miRge_build now requires Python 3.8. Do you happen to have an updated image with the correct dependencies? If so, I would greatly appreciate it if you could share it with me.

Thank you in advance for your help!

Best regards,
Tomoki

**Comment by arunhpatil** [comment]
Hi @TomokiMotegi,

The version seems ok. However, from your data, for a miRNA sequence has no match to its corresponding hairpin sequence. You can get an idea about the error [here](https://www.geeksforgeeks.org/re-matchobject-span-method-in-python-regex/). 

Please re-check your database input files. 

Thank you,
Arun.

**Comment by TomokiMotegi** [comment]
Hi Arun,

Thank you for your reply and the valuable information. I followed the steps to create a miRNA library for dogs using data from miRBase. Specifically, I extracted the Canis familiaris sequences from the "hairpin.fa" and "mature.fa" files in miRBase. Could you confirm if these steps are correct, or is there something I may have missed in the process?

I appreciate your guidance!

Best regards,
Tomoki

**Comment by arunhpatil** [comment]
Hi @TomokiMotegi,

The sequence from the miRbase fasta file has U's instead of T's. Can you try converting them first? 

Let me know how it goes. 

Thank you,
Arun.

**Comment by TomokiMotegi** [comment]
Hi Arun,

Thank you so much for your help! I was able to create an original canine library successfully.
It turns out the issue was caused by mismatched IDs between the hairpin miRNA file and the mature miRNA file.

I have one more question if you don’t mind. Should the chromosome names be consistent across the genome file, GFF, and cDNA FASTA? I prefer to use the Ensembl cDNA FASTA because it provides more annotations than UCSC. However, the chromosome naming conventions differ: our genome and GFF use "chrxx", while Ensembl uses "xx".

My test run seems to have worked well so far, but I want to confirm if this mismatch could potentially cause any issues. I'd greatly appreciate your advice if there’s anything I should be aware of.

Thank you again for your guidance!

Best,
Tomoki
